pkg/services/ngalert/api/generated_base_api_alertmanager.go

```diff
@@ -408,6 +414,16 @@ func (api *API) RegisterAlertmanagerApiEndpoints(srv AlertmanagerApi, m *metrics
 				m,
 			),
 		)
+		group.Post(
+			toMacaronPath("/api/alertmanager/grafana/config/history/{id}/_activate"),
+			api.authorize(http.MethodPost, "/api/alertmanager/grafana/config/history/{id}/_activate"),
```

Comment:
> I assume it's because "activate" refers to an action rather than a resource

Correct. @yuri-tceretian and I briefly discussed what format the endpoint should be in and came to this compromise. It has the benefits of mirroring the related endpoint `/api/alertmanager/grafana/config/history` and referencing the historical config by resource id. The drawback being the, somewhat strange, underscore action.

The other options we discussed were:

- POST `/api/alertmanager/grafana/config/api/v1/alerts` with `{id:<int>}`. The benefits of this being no extra endpoint needed. The drawbacks are that it conflates this new use-case with an existing core endpoint.
- POST `/api/alertmanager/grafana/config/1` with body `{ active : true }` or something similar. This doesn't have `_activate` but poses the question of what does this endpoint do if `active=false`.

I'm open to discussing alternatives if there's a compelling reason.